docs/details/backend.dox

```diff
@@ -16,6 +16,74 @@ will return in an exception.
 
 =======================================================================
 
+\defgroup unified_func_addbackendlibrary af_add_backend_library
+
+\brief Register an ArrayFire library located in an arbitrary path for use
+
+The custom library's path can be anywhere in the local file system, even outside
+of where the ArrayFire libraries are found by default (e.g. outside
+`LD_LIBRARY_PATH` in Unix and System or User Path in Windows). Use this in tandem
```

Comment:
`Use this in tandem` -> `To be used in tandem` with .. sounds better I think

Reply:
Thanks, but hmm I'm not sure. Changing it to that actually makes it an incomplete sentence. We could change it instead to `This is used in tandem...` so that it sounds less commanding and more neutral, if that's your concern.

Reply:
My concern is that the user should understand that both of those functions have to be used to change the active backend. Just calling add backend wouldn't suffice.

I don't think we added default behavior inside `af_add_backend_library` that switches current backend to newly added backend handle.

Reply:
Yes, you're right, calling `af_add_backend_library` does not switch the current backend. `af_set_backend_library` does that. I did change the wording here though to something similar to what @umar456 suggested in [this comment](https://github.com/arrayfire/arrayfire/pull/2514#discussion_r287619637)